### PR TITLE
Reference-to-View pass and comprehensive reference test suite

### PR DESCRIPTION
**Pull request by tbennun**
Implements a reference-to-view pass (converting references to views if they are only set to one particular subset). Also improves the simplify pipeline in the presence of Reference data descriptors and adds multiple tests that use references.

### CODE REVIEW

File: dace/transformation/passes/reference_reduction.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+            source = next(iter(reference_sources[cand]))
+            fsyms = source.subset.free_symbols
+            if not fsyms:
+                continue
+
+            for state, node in refsets[cand]:
+                # Check if any of the symbols is a scope symbol
+                entry = state.entry_node(node)
+                while entry is not None:
+                    if fsyms & entry.new_symbols(sdfg, state, {}):
```
Comment (phschaad) [inline dace/transformation/passes/reference_reduction.py:124]: Lines 124-129 are not hit by the tests, if it's easy to add one that causes these edge cases then it may be good to add one. Otherwise nevermind.

**Review by phschaad — APPROVED** [review]
Nice addition, LGTM
